ReactNativeClient/lib/services/InteropService_Importer_Md.js

```diff
@@ -54,21 +55,51 @@ class InteropService_Importer_Md extends InteropService_Importer_Base {
 		}
 	}
 
+	/**
+	 * Parse text for links, attempt to find local file, if found create Joplin resource
+	 * and update link accordingly.
+	 */
+	async importLocalImages(filePath, md) {
+		let updated = md;
+		const imageLinks = lo.uniq(extractImageUrls(md));
+		await Promise.all(imageLinks.map(async (link) => {
+			const attachmentPath = filename(`${dirname(filePath)}/${link}`, true);
+			const pathWithExtension =  `${attachmentPath}.${fileExtension(link)}`;
+			const stat = await shim.fsDriver().stat(pathWithExtension);
+			const isDir = stat ? stat.isDirectory() : false;
+			if (stat && !isDir) {
+				const resource = await shim.createResourceFromPath(pathWithExtension);
+				// NOTE: use ](link) in case the link also appears elsewhere, such as in alt text
+				const linkPatternEscaped = lo.escapeRegExp(`](${link})`);
+				const reg = new RegExp(linkPatternEscaped, 'g');
+				updated = updated.replace(reg, `](:/${resource.id})`);
+			}
+		}));
+		return updated;
+	}
+
 	async importFile(filePath, parentFolderId) {
 		const stat = await shim.fsDriver().stat(filePath);
 		if (!stat) throw new Error(`Cannot read ${filePath}`);
 		const title = filename(filePath);
 		const body = await shim.fsDriver().readFile(filePath);
+		let updatedBody;
+		try {
+			updatedBody = await this.importLocalImages(filePath, body);
+		} catch (error) {
+			// console.error(`Problem importing links for file ${filePath}, error:\n ${error}`);
```

Comment:
@laurent22 not sure what the best thing to do here is if not logging? My intent is just to ensure that should importing the linked images fail it will not cause the entire import of the file to fail.

Reply:
Normally we would log to the Logger class, which will take care of redirecting this to the right output (a file for example), but this is not setup for importer/exporter classes. We can't have direct console statements as that would break the GUI on CLI. I also don't like silenced exceptions but for now let's leave it out.